### DISCUSSION TITLE
GSoC 2024: Application Guide

### DISCUSSION THREAD

**Discussion by ashitaprasad** [foss42/apidash#111] [General]
GSoC 2024 [ **Application Guide** | [List of Ideas](https://github.com/foss42/apidash/discussions/112) ]

Hi 👋

Welcome to the API Dash ⚡️ project!
We're very glad that you are considering contributing to this project during Google Summer of Code 2024. 

## Pre-application steps

- Feel free to drop by our [Discord server](https://bit.ly/heyfoss) and wave a hi 👋. Discord is our primary mode to communication for this project. You can ask any question or interact with the mentors/admins anytime on this server. 
- Go through the [CONTRIBUTING GUIDE](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md) to learn how you can setup, run & test the project locally.
- Prior knowledge of Dart/Flutter is **not compulsory**. In case you want to get started, a good place to begin your learning journey is [here](https://flutter.dev/learn).
- Go through the existing [“Good First Issue”](https://github.com/foss42/apidash/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) to pick up an issue or raise a new issue and start interacting with the mentors to discuss your approach & how you can work on the issue. The objective is to help you get familiar with the project and the process of raising & working on a PR for API Dash. 

## Draft Application Review Process

1. Go through the pre-application steps if you are new to the project.
2. Create a new google doc, copy the below application template and fill it.
3. Share the google doc link in the comment below for review. 
 
**Like other open source projects, our application process is public so make sure the application link is not behind any login-wall and can be accessed publicly. Provide the following permission - "Anyone with access can View". Do not select "Commenter" or "Editor" role.**

## Application

Your GSoC application should comprise of the following sections:

### About 

1. Full Name
3. Contact info (email, phone, etc.)
6. Discord handle
7. Home page (if any)
8. Blog (if any)
9. GitHub profile link 
10. Twitter, LinkedIn, other socials
11. Time zone
12. Link to a resume (PDF, publicly accessible via link)

### University Info

1. University name
2. Program you are enrolled in (Degree & Major/Minor)
3. Year 
5. Expected graduation date

### Motivation & Past Experience

Short answers to the following questions (Add relevant links wherever you can):
1. Have you worked on or contributed to a FOSS project before? Can you attach repo links or relevant PRs? 
2. What is your one project/achievement that you are most proud of? Why?
3. What kind of problems or challenges motivate you the most to solve them?
4. Will you be working on GSoC full-time? In case not, what will you be studying or working on while working on the project?
6. Do you mind regularly syncing up with the project mentors?
7. What interests you the most about API Dash?
8. Can you mention some areas where the project can be improved?

### Project Proposal Information

This section contains details about the project you want to propose in your GSoC application. You can check out the potential list of projects here or propose a new one. 

1. Proposal Title
2. Abstract: A brief summary about the problem that you will be tackling & how.
3. Detailed Description
4. Weekly Timeline: A rough week-wise timeline of activities that you would undertake.

### Any Questions?

Please feel free to drop a comment below or drop a message on our Discord server.

**Comment by shadan1234** [comment]
Excited to contribute

**Reply by animator** [reply]
🧑🏻‍💻🚀

**Comment by ashutosh7i** [comment]
A project of my interest🥳,
Excited✨

**Reply by animator** [reply]
🧑🏻‍💻🚀

**Comment by akshayjadhav2002** [comment]
Inspired by the project want to contribute

**Reply by animator** [reply]
🧑🏻‍💻🚀

**Comment by thesakshidiggikar** [comment]
where we can submit application proposal?
@animator

**Reply by ashitaprasad** [reply]
Applications will be open from 18th March - https://summerofcode.withgoogle.com/programs/2024

**Comment by MUDITJAINN** [comment]
I am glad I landed here and looking forward to contributing.

**Reply by ashitaprasad** [reply]
🧑🏻‍💻🚀

**Comment by Narasimha677** [comment]
Looking Forward to contributing because it matches my intrests and skills and projects inspires me a lot

**Reply by animator** [reply]
Sure 🧑🏻‍💻🚀

**Comment by chiragtyagi2003** [comment]
Having Industry Level Experience with Flutter and Dart and leading my team to develop and deliver a product to the market. I think my skillset matches with the project. Looking forward to contributing. A little guidance on how to connect with the mentors and discuss new ideas would be helpful. Thanks

**Reply by animator** [reply]
Hi Chirag,
Would definitely love to the see the product that you & your team are currently building using Flutter.

**Comment by chiragtyagi2003** [comment]
Hi Ankit
The product that we built has been published on the play store. It was
under my tenure of an internship where I was a flutter developer as well as
a team lead.

Here's the link:
https://play.google.com/store/apps/details?id=com.pregamate.mobile_application

I would be glad to discuss it with you if you could find time.

**Reply by animator** [reply]
Nice Chirag!

**Comment by chiragtyagi2003** [comment]
Thanks a lot Ankit! Would love to know a bit more about your expectations
for your project and deliverables. I would be grateful for a little
guidance.

**Reply by ashitaprasad** [reply]
Hi Chirag, 
You can check out the list of projects here https://github.com/foss42/apidash/discussions/112 and the linked issues within each project to get started.

**Reply by chiragtyagi2003** [reply]
Alright, Thanks Ashita, I will get started and update you soon.

**Comment by Shrutim1505** [comment]
project seems interesting. I would be delighted to contribute in it. Looking forward to contribute in it further.

**Reply by ashitaprasad** [reply]
Sure 🧑🏻‍💻🚀

**Comment by suyashmadhesia** [comment]
Hi, I am interested in contributing to this project. Can I still apply even if I've already graduated?

**Reply by ashitaprasad** [reply]
Yes you can apply @suyashmadhesia

**Comment by apoorvdwi** [comment]
Hey @ashitaprasad @animator, I would love to have some feedback on my proposal
Here is the link for it - [GSoC 2024 API Dash Proposal](https://docs.google.com/document/d/1FEISZeJn2s5rWNVF6EE3a4pFNE8hR366erUpCa7CGs0/edit?usp=sharing)

**Reply by ashitaprasad** [reply]
@apoorvdwi Describe in detail the code generators and the libraries that you will be covering.
Also mention the length of the project and the detailed breakup of the weekwise plan for implementation of the code generators.

**Reply by apoorvdwi** [reply]
Got it, will do that. 
@ashitaprasad Apart from issues that have been raised already can I add more to the list for which issue is not raised yet?

**Reply by ashitaprasad** [reply]
@apoorvdwi Yes, you can do that

**Reply by apoorvdwi** [reply]
Hey @ashitaprasad I have added the project length and the week-wise plan for implementation. It is ready for further feedback (if any).

**Reply by ashitaprasad** [reply]
You can go ahead and submit your application in the GSoC portal.

**Comment by opxdelwin** [comment]
Hey @ashitaprasad @animator, please look into this draft proposal for [GSoC '24 @ API Dash Proposal (Mobile Support)](https://docs.google.com/document/d/1IU1UahUcsDsyG9swPFo-Vn-AtnCr05FI9lgCyQXMExI/edit?usp=sharing), and [API Dash Proposal GSoC '24 (Import Export)](https://docs.google.com/document/d/1EJCV1Jyy6CGd5tcIpmE4dIHsWwW1CUJSzURVy12buGk/edit?usp=sharing)

Awaiting valuable feedbacks.

Note: 
- Analytics mentioned in proposal is subject to approval of maintainers (specifics mentioned [here](https://discord.com/channels/920089648842293248/940195305620664330/1220694325000929300))

**Reply by opxdelwin** [reply]
Added preview images, ready for review!

**Reply by ashitaprasad** [reply]
@opxdelwin Review feedback:
Week 2 - What do you mean by sharing landing page?
Week 3 - No, we cannot add Google Analytics or any other analytics
You can make the above changes and revise your proposal.

**Reply by opxdelwin** [reply]
Analytics, noted.
Sharing: basically the user could share the app with their connections (here landing page --> https://apidash.dev/ ) so the others could use it as well.

**Reply by opxdelwin** [reply]
Hey I've incorporated few changes. Please do have a look. Also I'd appreciate any feedback on the UI

**Reply by ashitaprasad** [reply]
Looks good. As you can submit 3 proposals, you can submit both, but if you are going ahead with one then import-export project can be submitted.

**Reply by opxdelwin** [reply]
Great. I'd have preferred Mobile implementation. Is there anything I could make my proposal better? Else I'll move forward with `import-export`

**Reply by ashitaprasad** [reply]
In that case you can go ahead and submit both.

**Reply by opxdelwin** [reply]
Yes, I will go ahead and submit in the portal, please do drop feedback if there's any space for improvement.

**Reply by opxdelwin** [reply]
I have updated `import-export` proposal with implementation research as well, please do give a look when available 😊

**Comment by siddhardh-7** [comment]
Hello @ashitaprasad @animator,
I have drafted a [GSoC '24 Proposal @ API Dash](https://docs.google.com/document/d/1Y77aqrCqI67x2qtQgrU1m8G1eTRhfm6qwJfNC4U15u4/edit?usp=sharing) for integrating gRPC support into ApiDash and would greatly appreciate your feedback. Could you please take a moment to review it and share your thoughts?

**Reply by animator** [reply]
@siddhardh-7 Your document has been modified by a troll. Please take a look at it and set the right permission to "Anyone with access can View" as shown below:


![Screenshot 2024-03-20 at 2 17 26 AM](https://github.com/foss42/apidash/assets/615622/550a0e4b-ccc5-4f21-8362-6913dca143d9)

**Reply by siddhardh-7** [reply]
@animator @ashitaprasad, can you please review the updated link and provide feedback?

**Reply by ashitaprasad** [reply]
@siddhardh-7 Your resume is not publicly visible, it's behind a login wall. Add a google docs link and make sure that it is publicly accessible.
Please split the timeline on a weekly basis and not biweekly basis. Also, research about gRPC  and add more specific details regarding what you expect to achieve, how will you achieve it and what the end result will look like.

**Reply by siddhardh-7** [reply]
@ashitaprasad @animator Now my resume is now publicly accessible.  I have split the timeline every week and included more specific details on what I expect to achieve, and how I plan to achieve it. And proposal is ready for further feedback.

**Reply by ashitaprasad** [reply]
You can go ahead and submit it in the GSoC portal.

**Comment by YoussefMoRabie** [comment]
Hello @ashitaprasad  @animator ,
I am very passionate about API DASH and I like more than one project. Can I apply to two different ones?
If this is possible, is the introductory part in both proposals similar?

**Reply by animator** [reply]
You can put **Project Proposal Information 1** & **Project Proposal Information 2** in the same proposal for now.
After review, we can select the best out of these two and go forward with it.

**Comment by sixtusagbo** [comment]
Here's a draft of my proposal: [GSOC 2024 Proposal](https://docs.google.com/document/d/1V1_AZmxDCVh4MJaGRfKgIyyyXEZMNH8wIThyFEzZVPw/edit?usp=sharing)

I'll also like to contribute for the improvements I suggested when I'm done with the project I proposed.

I'm patiently awaiting your thoughts on it.

**Reply by ashitaprasad** [reply]
@sixtusagbo Please describe the deliverables weekwise in more detail instead of clubbing it in 3 weeks bucket.

**Reply by sixtusagbo** [reply]
Okay

**Reply by sixtusagbo** [reply]
@ashitaprasad I have updated my proposal, I await your review and feedback on it.

**Reply by ashitaprasad** [reply]
Now it looks good. You can go ahead and submit it in the GSoC portal.

**Comment by madhupashish** [comment]
I was a contributer last year 
this year i want this project 
<img width="861" alt="Screenshot 2024-03-20 135430" src="https://github.com/foss42/apidash/assets/119279720/5a45684f-da84-493e-ab50-cc863b217c14">

**Reply by ashitaprasad** [reply]
@madhupashish Glad to see your enthusiasm. You can share a proposal for review, based on the guidelines mentioned above.

**Comment by Drakennnnn** [comment]
Hi @ashitaprasad @animator, 
This is the application and draft proposal for GSoC - [GSoC '24 API DASH](https://docs.google.com/document/d/1ZJu_kVi-VMYzV0FkUY3uj8arQuc8pMxnBVfcO2vt1wM/edit?usp=sharing)
Kindly review and let me know your thoughts.
Thank You.

**Reply by ashitaprasad** [reply]
@Drakennnnn Project Proposal 1 feedback:
You cannot spend first two weeks just on the research .. it has to be done before the start of the project.
Describe the deliverables in more detail and add more specific details regarding what you expect to achieve, how will you achieve it and what the end result will look like.

Project Proposal 2 feedback :
Weekwise split is very vague. You need to get into the specifics and explain it in a better and more detailed way.

**Reply by ashitaprasad** [reply]
Once you have made the changes you can go ahead and submit it in the GSoC portal.

**Reply by Drakennnnn** [reply]
@ashitaprasad I have made the changes, took some time and gave detailed description for each week, also removed the second proposal as I can see others working on a similar one. Prioritized my first proposal to work more effeciently. Please let me know your feedback.

**Reply by animator** [reply]
As mentioned by @ashitaprasad you can go ahead and submit as you have made the required changes.

**Comment by mdrazak2001** [comment]
Hi @ashitaprasad @animator, here is my draft proposal [API-Dash_GSOC-24_Proposal_Mohammed-Razak](https://docs.google.com/document/d/1yhQrD264wNk_VOimXq_umKSgC88x5qF4fZcEQxaPKEI/edit)
Kindly have a look and please share your feedback.
Thanks!

**Reply by ashitaprasad** [reply]
@mdrazak2001 
Split the deliverables weekwise, instead of clubbing them in 2 weeks bucket.
Do some research and identify more content types for which you can build response previewer in this project.
Week 3 and 4 - Specify in detail what UI components you will be using. 
In video enhancements section, video compression is not a priority. 
Week 5 and 6 - How are the UI components mentioned here different from week 3 & 4. Explain in detail. Also, feedback with mentors will happen regularly and not only in week 5/6. So, you can remove that point.

Overall the timeline is very vague and needs much more detail.

**Reply by mdrazak2001** [reply]
Hey, I've updated the proposal accordingly. Please do have a look.

**Reply by ashitaprasad** [reply]
You can go ahead and submit your proposal in the GSoC portal.

**Reply by mdrazak2001** [reply]
Sure, do I select the project size to be medium since I intend to work over a period of 12 weeks (~175hrs) for my proposed project.

**Reply by animator** [reply]
medium

**Comment by YoussefMoRabie** [comment]
Hi @ashitaprasad @animator,
I'm excited to share with you the application and draft proposal for GSoC 24- [GSoC 24 API DASH | Yousef Rabia Proposal](https://docs.google.com/document/d/1Byy_zTYYNTJgmkijAMO3TUouNG4q2o-AUEO2eyuuyj4/edit)
Your feedback and thoughts would be greatly appreciated.

I plan to include additional project information soon, considering that GSoC typically accepts three proposals from each applicant.

Thank You.

**Reply by ashitaprasad** [reply]
@Yousef-Rabia Your resume is behind a login wall. So, please make sure it is publicly accessible.
In the `Project Proposal Information` -> `Detailed Description` section, highlight your findings from the market research you have conducted.
Split the deliverables weekwise, instead of clubbing them in 2 weeks bucket. Describe the deliverables in more detail and add more specific details regarding what you expect to achieve, how will you achieve it and what the end result will look like.

**Reply by YoussefMoRabie** [reply]
@ashitaprasad I update my resume accessibility

**Reply by YoussefMoRabie** [reply]
@ashitaprasad @animator 
I hope you're doing well! I've made updates to my GSoC application based on your previous feedback, particularly on the timeline and detailed description sections. Can you take another look?

**Note** : I've adjusted my timeline to fit the 12-week duration of the Google Summer of Code program. However, I think I can complete the project sooner due to dedicated focus on API DASH development. 

**Your insights are greatly appreciated!**

**Reply by ashitaprasad** [reply]
Thanks for making the changes. You can go ahead and submit your proposal in the GSoC portal.

**Comment by matte23n** [comment]
Hi @ashitaprasad @animator,
I'm very glad to share my [Proposal](https://docs.google.com/document/d/1iz_faeWT56M16n6FoNqx5vrJtq7-x7mfYaH_aDeMfGI/edit?usp=sharing) to work on the project regarding the implementation of GraphQL Support in API Dash.

Waiting to hear your thoughts about it.
Thanks!

**Reply by ashitaprasad** [reply]
@matte23n Split the deliverables weekwise, instead of clubbing them in 2 weeks bucket. Also, research about GraphQL and add more specific details regarding what you expect to achieve, how will you achieve it and what the end result will look like.

**Reply by matte23n** [reply]
Hi @ashitaprasad 
I've made updates based on your feedback. Can you take another look and let me know if there's something else I can still improve?

Your feedbacks are really appreciated!

**Reply by ashitaprasad** [reply]
You can go ahead and submit your proposal in the GSoC portal.

**Comment by ashutosh7i** [comment]
Hello @ashitaprasad & @animator ;
i am very excited to share [my GSOC24 proposal](https://docs.google.com/document/d/10OfXIaWamf-sYTGGdraD6lkuGsP0HLcacikCLZaYyn8/edit?usp=sharing) for Realtime API Testing Support project to ApiDash⚡

Kindly review and please share your valuable feedback.
Thanks You.

**Reply by ashitaprasad** [reply]
@ashutosh7i The deliverables mentioned are currently very vague. Research more about Websocket and MQTT and add more specific details regarding what features you will build, how will you build them and what the deliverables will look like.

**Comment by mmjsmohit** [comment]
Greetings @animator @ashitaprasad!
I would like to share my [GSoC Proposal](https://docs.google.com/document/d/1Bijkq7E08If85g3E7gm2gyatCYVOKBConGMf0nR7VNM/edit?usp=sharing) for the project: **"Realtime API Testing - WebSocket, MQTT and SSE (Server Sent Events)"**

Looking forward to your guidance and reviews.

Thanks in advance!

**Reply by ashitaprasad** [reply]
@mmjsmohit You can remove the week 1 things as you are supposed to do your research prior to the start of the project. For the weekwise plan go into more specifics like what features you will build, how will you build them and what will the end outcome look like. Also, project length should be of 12 weeks.

**Reply by mmjsmohit** [reply]
Greetings @ashitaprasad!

Since I have decided to include an additional idea in my proposal, I would greatly appreciate your valuable feedback at your convenience.

[Link to GSoC Proposal](https://docs.google.com/document/d/1t-n1sG2lz6bL2CEwDro0Q1yg1ohJg1gOQIpG552W480/edit?usp=sharing) for the project **"API Testing Suite, Env. Manager, Collection Runner & Monitor"**. Please accept my apologies for any inconvenience caused by the timing.

Thanks in advance!

**Reply by animator** [reply]
Looks good @mmjsmohit 👍
Please go ahead and submit it.

**Comment by PratyushChauhan** [comment]
Hello @animator @ashitaprasad , 
I have made a project proposal for Idea 9: API Workflow Builder.
I have gone through the relevant issues and set up the dev environment locally.

Please help me refine it before submission, your review would be very helpful!

Proposal: https://docs.google.com/document/d/1F7nylj_3tsTRDBN6b98xMafCnolpkxmrfLDTPz0Fk3Y/edit

**Reply by ashitaprasad** [reply]
@PratyushChauhan Milestones for the completion of the project will have to be decided during the project proposal phase and not during the community bonding period. Also, deliverables for mid term and end evaluations are closed during the project proposal phase and not during the evaluations. Explain your weekly deliverables clearly especially from week 7 to 12. In week 7 explain more about the performance generator tool.

**Reply by PratyushChauhan** [reply]
thanks for the feedback @ashitaprasad .

I am implementing the changes you have suggested.

Please let me know what metrics would you like to include in the API performance report aside from the basics:
- Memory Usage.
- Time to completion.
- Time between requests.

I am also implementing milestones as such, do let me know if anything is amiss:

- **Workflow editor window:** The user should be able to create an api request with relevant data in the form of a draggable box inside a workflow editor.

- **Request chaining:** Users should also be able to chain/connect different api call boxes.
- **Workflow execution:** Users should be able to run their api workflow via a button press.
- **Performance report generation:** The users should be able to view a performance report which will include:
  -     Memory Usage.
  -     Time to completion.
  -     Time between requests.

**Reply by PratyushChauhan** [reply]
The Timeline has also been updated in the document for your review @ashitaprasad .

https://docs.google.com/document/d/1F7nylj_3tsTRDBN6b98xMafCnolpkxmrfLDTPz0Fk3Y/edit

**Reply by PratyushChauhan** [reply]
Do let me know if anything else needs to be addressed, I'll work on it right away @ashitaprasad .

**Reply by ashitaprasad** [reply]
You can go ahead and submit your proposal in the GSoC portal.

**Comment by zenoR-git** [comment]
Greetings @animator @ashitaprasad
I am excited to share my [GSoC24 Project proposal](https://docs.google.com/document/d/18Fuu7i8WOdpecbeWqyeXU3DfZ5mExV4AqT15T-XuQZE/edit?usp=sharing) on **Idea 5: Setting up build process for various Linux packaging formats** and **Idea 8: Code Generator.**

Looking forward to your guidance and reviews.

**Reply by ashitaprasad** [reply]
@zenoR-git Remove buffer weeks from the proposal.

**Reply by zenoR-git** [reply]
@ashitaprasad I have updated by draft proposal as per you review. Should I go forward with [this](https://docs.google.com/document/d/18Fuu7i8WOdpecbeWqyeXU3DfZ5mExV4AqT15T-XuQZE/edit?usp=sharing)?

**Reply by ashitaprasad** [reply]
You can go ahead and submit your proposal in the GSoC portal.

**Comment by nikkivirtuoso** [comment]
Greetings @animator @ashitaprasad
I am excited to share my [GSoC24 Project proposal](https://docs.google.com/document/d/15hlpBLSbenHIWLt-50r7x3U9eWv_P4QNIqWBHrv1xHs/edit?usp=sharing) on Idea 8: Code Generator.

Looking forward to your guidance and reviews.

**Reply by ashitaprasad** [reply]
@nikkivirtuoso Explain the deliverables in detail weekwise for the entire duration of the project.

**Reply by nikkivirtuoso** [reply]
@ashitaprasad ma'am done with the task please have a look

**Reply by nikkivirtuoso** [reply]
@ashitaprasad @animator do I need to change anything ?

**Reply by ashitaprasad** [reply]
You can go ahead and submit your proposal in the GSoC portal.

**Comment by PratyushChauhan** [comment]


**Reply by animator** [reply]
@PratyushChauhan We already have your proposal review thread in this discussion - https://github.com/foss42/apidash/discussions/111#discussioncomment-8880481 .. please stick to this thread for any updates and do not create a new thread.

**Reply by PratyushChauhan** [reply]
Alright

**Comment by Tanish2002** [comment]
Hey @ashitaprasad @animator, A bit late 😅 but I would also love to have some feedback on my proposal
Here is the link for it - [GSoC 2024 API Dash Proposal](https://docs.google.com/document/d/1cLz05XdPuoFZNftFZOkrjsbpi-eu4PN-q7lolpTkedk/edit?usp=sharing)

**Reply by ashitaprasad** [reply]
Let us know when the final proposal is ready for review.

**Reply by Tanish2002** [reply]
@ashitaprasad ready for review!

**Reply by animator** [reply]
@Tanish2002 looks good ... you can go ahead and submit it.

**Comment by MohammedAshraf19** [comment]
Hey @animator @ashitaprasad 
I'm very glad to share my [Proposal](https://docs.google.com/document/d/1lNji0C9UTgKXPvCosqeceFDCPo-OGC2FYWJChSTwjAQ/edit?usp=sharing) 
Kindly have a look and please share your feedback.
Thanks!

**Reply by ashitaprasad** [reply]
- Designing should not take so much time.
- Explain the work you will do in week 9-12.
- Also, explain the end goals & deliverables.
Once you have made the changes, you can go ahead and submit your proposal in the GSoC portal.

**Reply by MohammedAshraf19** [reply]
hi @animator @ashitaprasad 
I'm ready to submit my proposal so
what is project size of  Android/iOS support issue?

**Reply by ashitaprasad** [reply]
Medium

**Comment by alainjr10** [comment]
Hey @animator , @ashitaprasad 
I'm delighted to share my draft proposal here. Hoping to get some feed back soon. [GSoc Proposal](https://docs.google.com/document/d/1rc46CQv4YaHluS0326qNgREHRMTN8hgqMQ-QHCJu2AQ/edit?usp=sharing)

**Reply by ashitaprasad** [reply]
- Your resume is behind a login wall
- Correct the formatting of your application. There are duplicate pages.
Once you have made the changes you can go ahead and submit your proposal in the GSoC portal.

**Reply by alainjr10** [reply]
Hey @ashitaprasad . This is a strange one and sounds weird perhaps, but I can't seem to be able to upload my proposal on the site. I've exported several times as pdf, but when i try uploading, I get the error, something went wrong, try another file.
When i try uploading a random pdf file, it uploads perfectly.

**Reply by animator** [reply]
Create a new file. export as PDF & try uploading.

**Comment by DenserMeerkat** [comment]
Hey mentors,

I would like to share my draft [GSoC Proposal](https://docs.google.com/document/d/1AgaL8B5n9FkH0NRQL-W9IUjIpQ0QmYkZrb2Rezxm6Gg/edit?usp=sharing) for Android/iOS support. 
Your feedback would be greatly appreciated. Thank you for your time.

**Reply by ashitaprasad** [reply]
Let us know when the proposal is ready for review.

**Reply by DenserMeerkat** [reply]
The proposal is ready for review 👍

**Comment by abhishekkumar11724** [comment]
Hey @animator and @ashitaprasad,

I'm excited to have stumbled upon this repository and am eager to contribute. I've put together a proposal and would greatly appreciate your feedback. Looking forward to hearing from you soon!
[Proposal](https://docs.google.com/document/d/1RwgWz5tiuvetBncmWuk4mZ4p7MLoAlpIGQyuiUbR18k/edit?usp=sharing)
Thank you for your time and consideration.

**Reply by ashitaprasad** [reply]
As you have already submitted your proposal, no more changes is required.

**Reply by abhishekkumar11724** [reply]
If there is anything you require me to change I can do so, 
I would really appreciate any feedback on my proposal.

**Comment by AcousticDeveloper** [comment]
Hey @animator and @ashitaprasad,

I would like to share the draft of my [GSoC proposal](https://docs.google.com/document/d/1wUIPFu0EyN4GybBC61f9Za7Au_9gAK9f6wxABcVnj3Q/edit?usp=sharing) and would greatly appreciate the feedbacks from you.

Thank you for your consideration.

**Reply by ashitaprasad** [reply]
You can go ahead and submit your proposal in the GSoC portal.

**Reply by AcousticDeveloper** [reply]
![image](https://github.com/foss42/apidash/assets/161834431/a02d21dd-1bbd-4f22-bca8-216e22467323)

@ashitaprasad During submission of the proposal, there is a field which asks for the project size which must match the size organization expects. In the project description, the size is written either medium or large based on the number of code generators to be added. My proposal is about adding 18 code generators for various programming languages and 3 more related to front end frameworks. 

So can you please suggest if you have any expected size in mind?

**Reply by animator** [reply]
medium

**Comment by madhupashish** [comment]
Hi [@ashitaprasad](https://github.com/ashitaprasad) [@animator](https://github.com/animator),
This is the application and draft proposal for GSoC - [GSoC '24 API DASH](https://docs.google.com/document/d/1e-Aop4GWtBfp1Q5KeB9ax-Lrm40aeD9R/edit?usp=sharing&ouid=109053269320205786536&rtpof=true&sd=true)
Kindly review and let me know your thoughts.
Thank You.

**Reply by ashitaprasad** [reply]
As you have already submitted your proposal, no more changes is required.

**Comment by lamergameryt** [comment]
Hi @ashitaprasad & @animator,

Could you please review my [draft proposal](https://docs.google.com/document/d/11CATRzME_Yv1Q5AZBVUzSVm9oTY_NmYdrQ8Ba30sQpA/edit?usp=sharing) for GSoC 24? Let me know if I should make any changes or if I can go ahead and submit my proposal.

Thanks!

**Reply by animator** [reply]
Looks good, you can go ahead and submit it 👍

**Reply by lamergameryt** [reply]
Thank you, @animator!

**Comment by mohnishdeshpande** [comment]
Hi @ashitaprasad @animator!

Please have a look at my proposal draft [here](https://docs.google.com/document/d/1kMezUZMXacE6sk3NZZ1N5wS8g9W22MgfMO9MdPVIW_0/edit?usp=sharing)
Any feedback is appreciated.

Thanks!

**Reply by animator** [reply]
@mohnishdeshpande you can go ahead and submit the proposal.

**Comment by m0hit-kumar** [comment]
Hi @ashitaprasad  @animator 

Please have a look at my proposal draft [here](https://docs.google.com/document/d/1PT3aAyBA0krxJVBLLtP8F60TjWal2rx-VbEkx8u9JuE/edit?usp=sharing)

Any change and feedback are most welcome

Thanks!

**Reply by ashitaprasad** [reply]
As you have already submitted your proposal, no more changes is required.

**Comment by sameer1098alam** [comment]
i am intersted to contribute

**Comment by Mayank-pratap-singh10** [comment]
highly excited to contribute

**Reply by animator** [reply]
@Mayank-pratap-singh10 this is 2024 thread 
2025 link - https://github.com/foss42/apidash/discussions/564
